packages/catalog-process/src/services/catalogService.ts

```diff
@@ -752,13 +740,10 @@ export async function deleteDraftDescriptorLogic({
   assertEServiceExist(eServiceId, eService);
   assertRequesterAllowed(eService.data.producerId, authData.organizationId);
 
-  const descriptor = eService.data.descriptors.find(
-    (d: Descriptor) =>
-      d.id === descriptorId && d.state === descriptorState.draft
-  );
+  const descriptor = retrieveDescriptor(descriptorId, eService);
```

Comment:
Refactor and minor change. Previously, the `deleteDraftDescriptor` operation resulted in `eServiceDescriptorNotFound` even in case the descriptor was there but not in draft state. With this change, we could distinguish between `eServiceDescriptorNotFound` and `notValidDescriptor`. The second is checked at the lines below.